lib/splitclient-rb/clients/split_client.rb

```diff
@@ -62,22 +63,22 @@ def get_treatment(key, split_name, attributes = nil, split_data = nil, store_imp
       end
 
       start = Time.now
-      treatment_with_label = { label: Engine::Models::Label::EXCEPTION, treatment: Treatments::CONTROL }
+      treatment_label_change_number = { label: Engine::Models::Label::EXCEPTION, treatment: Treatments::CONTROL }
```

Comment:
@adilaijaz Let's divide possible exceptions into two sets:
1. Those exceptions which are not related to the successful delivery of impressions
2. Those which are related, i.e. if there was an exception while fetching split data, obviously, SDK can't report anything reasonable as impression

So, the answer to your initial question is: if we are talking about exceptions from the first set, then, even if an exception was thrown we will deliver an impression. But, we can't send impression if we're dealing with the exception of the second set.

If my answer still makes no sense, maybe I did not understand your question correctly.